diffyscan/utils/hardhat.py

```diff
@@ -0,0 +1,77 @@
+import os
+import subprocess
+import signal
+
+from urllib.parse import urlparse
+from .logger import logger
+
+
+class Hardhat:
+    sub_process = None
+    TIMEOUT_FOR_INIT_SEC = 5
+
+    def __init__(self):
+        pass
+
+    def is_port_in_use(self, parsed_url) -> bool:
+        import socket
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            return s.connect_ex((parsed_url.hostname, parsed_url.port)) == 0
+
+    def start(self, main_config_relative_path: str, binary_config):
+        parsed_url = urlparse(binary_config["local_RPC_URL"])
+        hardhat_config_relative_path = Hardhat.get_config_path(
+            os.path.dirname(main_config_relative_path),
+            "hardhat_configs",
+            binary_config["hardhat_config_name"],
+        )
+        local_node_command = (
+            f"npx hardhat node --hostname {parsed_url.hostname} "
+            f"--port {parsed_url.port} "
+            f"--config {hardhat_config_relative_path} "
+            f"--fork {binary_config["remote_RPC_URL"]}"
```

Comment:
@tamtamchik after adding sepolia fork deploy stopped to work, this problem hasn't solved yet

Reply:
@Alexvozhak thank you for the update